drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -444,8 +442,51 @@ function DragAndDropTemplates(configuration) {
     };
 
     var mainTemplate = function(ctx) {
-        var problemTitle = ctx.show_title ? h('h3.problem-title', {innerHTML: ctx.title_html}) : null;
+        var progress_id = configuration.url_name + '-problem-progress';
+        var problemTitle = null;
+        if (ctx.show_title) {
+            problemTitle = h('h3.problem-title', {
+                innerHTML: ctx.title_html,
+                attributes: {'aria-describedby': progress_id}
+            });
+        }
         var problemHeader = ctx.show_problem_header ? h('h4.title1', gettext('Problem')) : null;
+        // Formats a number to 4 decimals without trailing zeros
+        // (1.00 -> '1'; 1.50 -> '1.5'; 1.333333333 -> '1.3333').
+        var formatNumber = function(n) { return n.toFixed(4).replace(/\.?0+$/, ''); };
+        var is_graded = ctx.graded && ctx.max_score > 0;
+        var progress_template;
+        if (ctx.grade !== null) {
+            if (is_graded) {
+                progress_template = ngettext(
+                    '{earned}/{possible} point (graded)',
+                    '{earned}/{possible} points (graded)',
+                    ctx.max_score
+                );
+            } else {
+                progress_template = ngettext(
+                    '{earned}/{possible} point (ungraded)',
+                    '{earned}/{possible} points (ungraded)',
+                    ctx.max_score
+                );
+            }
+            progress_template = progress_template.replace('{earned}', formatNumber(ctx.grade));
+        } else {
+            if (is_graded) {
+                progress_template = ngettext(
+                    '{possible} point possible (graded)',
+                    '{possible} points possible (graded)',
+                    ctx.max_score
+                );
+            } else {
+                progress_template = ngettext(
+                    '{possible} point possible (ungraded)',
+                    '{possible} points possible (ungraded)',
+                    ctx.max_score
+                );
+            }
+        }
```

Comment:
You're right, that function was getting way too long. I moved the progress template to a separate function.